operator/api/v1alpha1/plan_types.go

```diff
@@ -20,38 +20,96 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
-// PlanSpec defines the desired state of Plan
+// PlanSpec defines the Service Plan spec.
 type PlanSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// The name of the Offering this Plan belongs to.
+	Offering string `json:"offering"`
+	// A unique ID for the Plan to be used by OSBAPI. If not provided, a UUID v1 is auto-generated.
+	// +optional
+	ID string `json:"id,omitempty"`
+	// A description for the Plan.
+	// +optional
+	Description string `json:"description,omitempty"`
+	// The specification for how an Instance of the Plan should be provisioned.
+	Provisioning PlanProvisioningSpec `json:"provisioning"`
+	// The specification for how an Instance of the Plan should be bound.
+	Binding PlanBindingSpec `json:"binding"`
+}
+
+// PlanProvisioningSpec defines the Provisioning spec for a Plan.
+type PlanProvisioningSpec struct {
+	// Chart contains what chart to be used to provision an Instance of the Plan.
+	Chart PlanProvisioningChartSpec `json:"chart"`
+	// Values contains the configuration for validating user-provided provisioning properties as
+	// well as plan-specific default values and overrides.
+	Values PlanProvisioningValuesSpec `json:"values"`
+}
+
+// PlanProvisioningChartSpec defines the Chart spec for a Provisioning.
+type PlanProvisioningChartSpec struct {
+	// The URL for the Chart tarball.
+	URL string `json:"url"`
+}
+
+// PlanProvisioningValuesSpec defines the Values spec for a Provisioning.
+type PlanProvisioningValuesSpec struct {
+	// The JSON schema for validating user-provided properties that are passed to the Helm client as
+	// values for installing a Chart. The schema definition can be written as YAML or JSON.
+	Schema string `json:"schema"`
+	// The default values used to override the Chart defaults. The user-provided values can still
+	// override these values.
+	// +optional
+	Default string `json:"default,omitempty"`
+	// The static values applied on top of all other values used to enforce plan-specific
+	// configuration.
+	// +optional
+	Static string `json:"static,omitempty"`
+}
+
+// PlanBindingSpec defines the Binding spec for a Plan.
+type PlanBindingSpec struct {
+	// The specification of the desired behaviour of the binding credentials.
+	Credentials PlanBindingCredentialsSpec `json:"credentials"`
+}
+
+// PlanBindingCredentialsSpec defines the Credentials spec for a Binding.
+type PlanBindingCredentialsSpec struct {
+	// The container specification for the logic of binding an Instance of the Plan.
+	RunContainer PlanBindingCredentialsRunContainerSpec `json:"runContainer"`
```

Comment:
Would it make sense (or even possible) to reuse [`core/v1.Container`](https://v1-18.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#container-v1-core)?

Reply:
I'm not sure I want to expose the whole container spec to the user. This way I can be more opinionated on the container spec.